src/app/pages/quickMaps/pages/locationSelect/locationSelect.component.ts

```diff
@@ -36,58 +50,143 @@ export class LocationSelectComponent implements OnInit, AfterViewInit {
     }, 0);
   }
 
-  public initialiseMap(): void {
+  public goToCountry(id: string): void {
+    this.dictionaryService.getDictionary(DictionaryType.COUNTRIES).then((dict: Dictionary): void => {
+      const country = dict.getItem<CountryDictionaryItem>(id);
+      if (null != country && null != country.geoFeature) {
+        // this.map.fitBounds(country.geoFeature.geometry);
+        this.hoverHighlightFeature(country.geoFeature);
+      }
+    });
+  }
+
+  public getGeoJsonData(): void {
+    this.dictionaryService.getDictionary(DictionaryType.COUNTRIES).then((dict: Dictionary): void => {
+      const featureCollection: GeoJSON.FeatureCollection = {
+        type: 'FeatureCollection',
+        features: dict
+          .getItems<CountryDictionaryItem>()
+          .map((item) => item.geoFeature)
+          .filter((item) => null != item),
+      };
+      this.geojson = L.geoJSON(featureCollection, {
+        style: () => { // eslint-disable-line no-use-before-define
+          return {
+            fillColor: '#8a66ad',
+            fillOpacity: 0.1,
+            color: '#3a1d54',
+            opacity: 0.8,
+          };
+        }
+      });
+    });
+  }
+
+  public getselectedLayer(countryId: string): CountryDictionaryItem {
```

Comment:
I think this function needs a bit of tidying:
- It states it is returning a CountryDictionaryItem but it seems to return a map layer (L.layer?).
- There is no return statement for if this.geojson is falsey (null).  Suggest moving return statement to the very bottom of the function.
- You could add (null === country) to the start of the "if" on line 89 to stop it from still looking for the country after it's already found it, to improve performance, but the saving would be tiny.